apps/server/src/controllers/userControllers/getUserTransactionHistory.ts

```diff
@@ -2,15 +2,34 @@ import { Response } from 'express'
 import { RequestExt } from '../../middleware/authorization/authentication'
 import Transactions from '../../models/transactions'
 import { HTTP_STATUS_CODE } from '../../constants'
+import Wallets from '../../models/wallets'
+import Users from '../../models/users'
 
 export const getTransactionHistory = async (req: RequestExt, res: Response) => {
   try {
     const { _userId: userId } = req.body
-    const transactionHistory = await Transactions.findAll({
+
+    const userTransactions = await Transactions.findAll({
       where: { ownerId: userId },
       order: [['createdAt', 'DESC']],
       limit: 5,
+      include: [{ model: Wallets }, { model: Users }],
     })
+
+    // Extract relevant information for the response
+    const transactionHistory = userTransactions.map((transaction) => ({
+      id: transaction.id,
+      walletId: transaction.walletId,
+      ownerId: transaction.ownerId,
+      ownerName: transaction.name,
+      action: transaction.action,
+      type: transaction.type,
+      receiverId: transaction.receiverId,
+      senderId: transaction.senderId,
+      date: transaction.createdAt,
+      amount: transaction.amount,
+    }))
```

Comment:
let's get the most recent transactions within the week

so

Sun - Sat

Reply:
then maybe on request from the user, we send everything like asking for your bank statement.

Reply:
I think we need more info:

<img width="970" alt="Screenshot 2024-02-13 at 16 37 06" src="https://github.com/Ajo-Savings-Project/ajo-savings/assets/48376769/ee6c202b-0317-4942-b586-6b2d99d642b3">

according to the UI, clicking on the card will most likely bring a popup

we can talk more here

Reply:
ok